chore: make updates for innersource readiness (#3514)
salesforce/lwc: CONTRIBUTING.md

[eslint-integrations]: http://eslint.org/docs/user-guide/integrations
[integration-test-readme]: https://github.com/salesforce/lwc/blob/master/packages/@lwc/integration-tests/README.md
[integration-karma-readme]: https://github.com/salesforce/lwc/blob/master/packages/@lwc/integration-karma/README.md

## Getting your changes reviewed

**Potential pitfalls:** When submitting changes to the LWC framework, there are several important considerations to keep in mind to ensure that your changes are reviewed & accepted.

- The LWC framework makes strong guarantees about backwards compatibility to those that use our framework. Any changes that compromise these guarantees are unlikely to be accepted as-is.

> - If your change causes a test to fail in unit or integration tests, it is likely that your submission won't be accepted as-is.

Reviewer: Making it consistent with the line above
```suggestion
- If your change causes a test to fail in unit or integration tests, it is unlikely that your submission will be accepted as-is.
```

- Certain parts of the codebase are particularly susceptible to breaking changes. This includes the browser runtime (`engine-core` and `engine-dom`), as well as the compiler (`compiler`, `style-compiler`, `template-compiler`).

> - Some areas of the codebase are more less mature and undergoing active development. Areas related to SSR, including the server-side runtime (engine-server), fall under this category.

Reviewer 2: ```suggestion
- Some areas of the codebase are more less mature and undergoing active development. Areas related to SSR, including the server-side runtime (`engine-server`), fall under this category.
```

> - Some areas of the codebase are more less mature and undergoing active development. Areas related to SSR, including the server-side runtime (engine-server), fall under this category.

Reviewer 3: ```suggestion
- Some areas of the codebase are less mature and undergoing active development. Areas related to SSR, including the server-side runtime (engine-server), fall under this category.
```

Author: Fixed this manually (conflicted with another suggested edit).

> **Definition of done:** Especially when making large changes to the codebase, certain complementary work must be undertaken for the desired changes to be considered.  When making a submission please adhere to the following:

Reviewer 3: ```suggestion
**Definition of done:** Especially when making large changes to the codebase, certain complementary work must be undertaken for the desired changes to be considered. When making a submission please adhere to the following:
```
